### ISSUE TITLE
Image first loads older one when updated on the server

### ISSUE THREAD

**Issue by iosfitness** [DylanVann/react-native-fast-image#721]
**Describe the bug**
I am rendering a carousel of images in flat-list. Initially, when I uploaded the image on the server, the App fetched it properly and displayed on UI
Later, I updated the image content keeping the URL. Now, the image first uploads the older image as a placeholder and then shows the latest (actual) one.

Below is the code snippet :

```
import FastImage from 'react-native-fast-image';
import { createImageProgress } from 'react-native-image-progress';

```
```
// Wrap FastImage with react-native-image-progress.
const Image = createImageProgress(FastImage);
<Image
          source={{uri:formatSourceIfNeeded(item), 
            priority: FastImage.priority.normal}}
          style={styles.media}
          indicator={ProgressBar.CircleSnail}
          resizeMode={FastImage.resizeMode.cover}
          fallback={true}
          defaultSource={require('../resources/images/user.png')}
        />


export function formatSourceIfNeeded(str) {
  var tarea = str
    if (tarea !== null && tarea !== undefined && tarea.includes("/") && tarea.includes(".")) {
      if (tarea.indexOf("http://") == 0 || tarea.indexOf("https://") == 0) {
        return str
      }
    }
  return ""
}
```

```
const styles = StyleSheet.create({
 media: {
    width: '100%',
    height: 150,
    alignSelf: 'center'
  },
});
```

**To Reproduce**
Steps to reproduce the behavior if possible, or a link to a reproduction repo:
1. Freshly install the App
2. Click on any item from Listing page
3. Go on the Product Details page
4. See error https://drive.google.com/drive/folders/1BsY2om8r8OdU9_bP1deGzh1ebPWigaQT?usp=sharing

**Expected behavior**
The image should not show the previous cached image and instead load the latest one.

**Dependency versions**

- React Native version:    ` "react-native": "0.61.5"`
- React Native Fast Image version: ` "react-native-fast-image": "^8.3.2"`

I believe, I might need to update the **cache policy** and **priority** but just want to confirm if that would resolve my issue. 
Also, is there any support to add a **placeholder**?
Any suggestions to improve my approach are welcome. Thanks in advance

**Comment by LukaZagar1995** [comment]
source.cache?: enum
FastImage.cacheControl.immutable - (Default) - Only updates if url changes.
FastImage.cacheControl.web - Use headers and follow normal caching procedures.
FastImage.cacheControl.cacheOnly - Only show images from cache, do not make any network requests.

**Comment by iosfitness** [comment]
@LukaZagar1995 Thanks for the response. I tried FastImage.cacheControl.web but I still see the older image loading first and then the new one
I assume the issue is with HTTP headers. Could you please comment on that?

**Comment by LukaZagar1995** [comment]
Guessing because od default source. Try to remove it

**Comment by iosfitness** [comment]
@LukaZagar1995 Thanks a lot! for pointing that out. Apparently, that was the issue. 

For anyone looking to add **placeholder** please refer to my implementation. It seems to work with the latest version of fast-image

**Comment by LukaZagar1995** [comment]
Yeah but you don't actually use fast images because of the prop fallback that just transforms fastImage to regular image from react native, so it will always send request to server for image, it will not cache them.
